charts/spiffe-helper-csi-driver/templates/_spire-lib.tpl

```diff
@@ -0,0 +1,26 @@
+{{- define "spire-lib.registry" }}
+{{- if ne (len (dig "spire" "image" "registry" "" .global)) 0 }}
+{{- print .global.spire.image.registry "/"}}
+{{- else if ne (len (.image.registry)) 0 }}
+{{- print .image.registry "/"}}
+{{- end }}
+{{- end }}
+
+{{- define "spire-lib.image" -}}
+{{- $registry := include "spire-lib.registry" . }}
+{{- $repo := .image.repository }}
+{{- $tag := (default .image.tag .image.version) | toString }}
+{{- if eq (substr 0 7 $tag) "sha256:" }}
+{{- printf "%s/%s@%s" $registry $repo $tag }}
+{{- else if .appVersion }}
+{{- $appVersion := .appVersion }}
+{{- if and (hasKey . "ubi") (dig "openshift" false .global) }}
+{{- $appVersion = printf "ubi-%s" $appVersion }}
+{{- end }}
+{{- printf "%s%s:%s" $registry $repo (default $appVersion $tag) }}
```

Comment:
This is printf'ing things that were composed by printf'ing above.  Might be better to consolidate, such that we only have one printf involved in each "condition"